src/providers/mbed_provider/mod.rs

```diff
@@ -56,13 +57,17 @@ const SUPPORTED_OPCODES: [Opcode; 7] = [
     Opcode::ListOpcodes,
 ];
 
+const KEY_SLOT_COUNT: isize = 32;
+
 pub struct MbedProvider {
     // When calling write on a reference of key_id_store, a type
     // std::sync::RwLockWriteGuard<dyn ManageKeyIDs + Send + Sync> is returned. We need to use the
     // dereference operator (*) to access the inner type dyn ManageKeyIDs + Send + Sync and then
     // reference it to match with the method prototypes.
     key_id_store: Arc<RwLock<dyn ManageKeyIDs + Send + Sync>>,
     local_ids: RwLock<LocalIdStore>,
+    key_handle_mutex: Mutex<()>,
```

Comment:
Maybe it would be nice to add a comment on top of those lines to explain why the Mutex, Semaphore are needed. Just as an help for future code readers.

Reply:
Done